src/Socket.php

```diff
@@ -756,123 +748,65 @@ public static function showListForItem($item)
-        echo "<table class='tab_cadre_fixehov'>";
-        $header_begin  = "<tr>";
-        $header_top    = '';
-        $header_bottom = '';
-        $header_end    = '';
-
-        if ($canedit) {
-            $header_begin  .= "<th width='10'>";
-            $header_top    .= Html::getCheckAllAsCheckbox($massive_action_form_id);
-            $header_bottom .= Html::getCheckAllAsCheckbox($massive_action_form_id);
-            $header_end    .= "</th>";
-        }
-        $header_end .= "<th>" . __('Name') . "</th>";
-        $header_end .= "<th>" . __('Position') . "</th>";
-        $header_end .= "<th>" . SocketModel::getTypeName(0) . "</th>";
-        $header_end .= "<th>" . __('Wiring side') . "</th>";
-        $header_end .= "<th>" .  _n('Network port', 'Network ports', Session::getPluralNumber()) . "</th>";
-        $header_end .= "<th>" .  Cable::getTypeName(0) . "</th>";
-        $header_end .= "<th>" .  __('Itemtype') . "</th>";
-        $header_end .= "<th>" .  __('Item Name') . "</th>";
-        $header_end .= "</tr>\n";
-        echo $header_begin . $header_top . $header_end;
-
-        Session::initNavigateListItems(
-            "Socket",
-            //TRANS: %1$s is the itemtype name,
-                           //       %2$s is the name of the item (used for headings of a list)
-                                     sprintf(
-                                         __('%1$s = %2$s'),
-                                         $item->getTypeName(1),
-                                         $item->getName()
-                                     )
-        );
 
+        $entries = [];
+        $socket_form_url = self::getFormURL();
+        $socket = new Socket();
+        $networkport = new NetworkPort();
+        $cable = new Cable();
         foreach ($iterator as $data) {
-            $socket = new Socket();
             $socket->getFromDB($data['id']);
-            Session::addToNavigateListItems(get_class($socket), $socket->fields["id"]);
-            echo "<tr class='tab_bg_1'>";
-
-            if ($canedit) {
-                echo "<td width='10'>";
-                Html::showMassiveActionCheckBox(__CLASS__, $socket->fields["id"]);
-                echo "</td>";
-                echo "<td><a href='" . $socket->getFormURLWithID($socket->fields["id"])
-                                . "&amp;onglet=1'>" . $socket->fields["name"] . "</a></td>";
-            } else {
-                echo "<td>" . $socket->fields["name"] . "</td>";
-            }
 
-            echo "<td>" . $socket->fields["position"] . "</td>";
-            echo "<td>" . Dropdown::getDropdownName(SocketModel::getTable(), $socket->fields["socketmodels_id"]) . "</td>";
-            echo "<td>" . self::getWiringSideName($socket->fields["wiring_side"]) . "</td>";
-
-            $networkport = new NetworkPort();
+            $socket_name = $canedit ? "<a href='{$socket_form_url}?id={$data['id']}'>" . $socket->fields["name"] . "</a>" : $socket->fields["name"];
```

Comment:
On main branch, when HTML is build using string concatenation, it is necessary to escape the dynamic data manually:
```suggestion
            $socket_name = $canedit ? "<a href='{$socket_form_url}?id={$data['id']}'>" . htmlspecialchars($socket->fields["name"]) . "</a>" : htmlspecialchars($socket->fields["name"]);
```

Maybe it is necessary to handle the fact that the field could be nullable.

The same remark applies for the whole GLPI codebase on `main` branch.